app/src/main/java/com/example/testapp/data/Resource.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.testapp.data
+
+/**
+ * Network and cache data model
+ */
+sealed class Resource<out T : Any> {
+    class Data<out T : Any>(val data: T) : Resource<T>()
+    class Error(val exception: Throwable) : Resource<Nothing>()
+
+    // Function returns the state into string
+    override fun toString(): String {
+        return when (this) {
+            is Data<*> -> "Success[data=$data]"
```

Comment:
No hardcoded strings should be used. Please use string concatenation format form strings.xml